nexus/src/app/network_interface.rs

```diff
@@ -213,3 +214,64 @@ impl super::Nexus {
         }
     }
 }
+
+fn validate_transit_ips(ips: &[IpNet]) -> Result<(), Error> {
```

Comment:
I don't know the omicron codebase very well, but most of the logic here (checking for a valid unicast prefix + host bits being unset) seems like it would be useful in other places -- even maybe in maghemite.
Would it maybe make more sense to move this into a couple oxnet helper functions and use validate_transit_ips as a wrapper with context specific logic? (like checking if all the prefixes are the same AF)
e.g.
I could see both an IpNet.valid_unicast() and some kind of prefixes_overlapping() fn being useful in other places.

Reply:
This should now be updated to use the new methods from oxnet v0.1.1 -- let me know what you think. I ended up leaning towards keeping the address families we want to outlaw more explicit here rather than pushing them up in to something like an `is_unicast` method. The overlap checking looks vastly simpler now that oxnet offers it properly, which I'm happy with.

Reply:
Looks good! Only one last question: does the overlaps() method detect identical/duplicate prefixes?
e.g.
Would the find() call on line 248 return Some(T) if the two values were both 10.0.0.0/24?

Reply:
We do, this is captured in one of the integration tests: https://github.com/oxidecomputer/omicron/blob/6a086b336182ebe2a864654f5a7cdf6a2e204bbf/nexus/tests/integration_tests/instances.rs#L3271-L3290

Reply:
Ok, cool. I definitely missed that. LGTM!